lib/route.js

```diff
@@ -338,6 +338,18 @@ function buildRouting (options) {
     } else {
       runPreParsing(null, request, reply)
     }
+
+    if (context.onTimeout !== null) {
+      request.raw.socket.on('timeout', () => {
+        hookRunner(
+          context.onTimeout,
+          hookIterator,
+          request,
+          reply,
+          noop
+        )
+      })
```

Comment:
Unfortunately this is introduces a memory leak. The socket can be shared across multiple request (http keep-alive) and as a result this will leak. We need to make sure that this function is removed after the _raw_ response is completed.

Reply:
so instead of `noop` what about
```js
function clearTimeout(request,reply, listener) {
   request.raw.socket.removeListener('timeout', listener)
}
// OR
function clearTimeout(request,reply) {
   request.raw.socket.removeAllListeners()
}
```

Reply:
That won't work as the same keep alive socket could be used for _infinite_ request/response. I think you can do do:

```js
if (!socket._meta) {
  // only install once
  socket.on('timeout', handleTimeout)
}
socket._meta = { context, request, reply }

// at the top level
function handleTimeout () {
  const { context, request, reply } = this._meta
  hookRunner(
          context.onTimeout,
          hookIterator,
          request,
          reply,
          noop
        )
}
```

Reply:
Added the fix